types/vue.d.ts

```diff
@@ -2,17 +2,17 @@
  * Extends interfaces in Vue.js
  */
 
-import Vue, { ComponentOptions } from "vue";
+import { ComponentCustomOptions, ComponentCustomProperties } from "vue";
 import { Store } from "./index";
 
-declare module "vue/types/options" {
-  interface ComponentOptions<V extends Vue> {
+declare module "@vue/runtime-core" {
+  interface ComponentCustomOptions {
     store?: Store<any>;
   }
 }
 
-declare module "vue/types/vue" {
-  interface Vue {
+declare module "@vue/runtime-core" {
+  interface ComponentCustomProperties {
```

Comment:
Do we remove store type from instance? We have an issue about this at #994

Reply:
Ah didn't see that issue, but yea, it makes sense. Let me remove the declaration from `ComponentCustomProperties`. We can keep `ComponentCustomOptions` since it doesn't matter so much right...? (it wouldn't be used by users to check store types anyway).

Do you think it can be applied to Vue 3 as well? Or would it be breaking change...? 🤔

Reply:
> We can keep ComponentCustomOptions since it doesn't matter so much right...?

Yes, I think so.

> Do you think it can be applied to Vue 3 as well?

Do you mean Vuex 3? As it's technically breaking change, we should only apply it for Vuex 4.

Reply:
Got ya. I'll also put this to my todo list to add in docs as well. I'll update the PR 👍

Reply:
Updated! I've removed the type test where checking `this.$store` as well.